lib/serve/functions.js

```diff
@@ -0,0 +1,123 @@
+'use strict';
+
+var _ = require('lodash');
+var chalk = require('chalk');
+var path = require('path');
+var RSVP = require('rsvp');
+var getProjectId = require('../getProjectId');
+var utils = require('../utils');
+var parseTriggers = require('../parseTriggers');
+var EmulatorController;
+var controller;
+
+function _pollOperation(op) {
```

Comment:
Not exactly, the other pollOperation functions have a similar structure but have varying degrees of complexity in terms of what it logs, and how it can tell whether an operation is done. (In this case, it's an array of results, and the first element has a done property). In the deploy scenario, there's a body to the response, which has a done property.) Refactoring might be more complicated, and lead to harder to read code.